### PR TITLE
[SDPA-5959] Remove 'Add menu' button

### PR DESCRIPTION
**Pull request by christopher-hopper**
### Changes

1. Add the "Create Menus Permission" module
2. Install module with default permissions to remove 'Add menu' task for non-administrators

Jira Ticket: https://digital-engagement.atlassian.net/browse/SDPA-5959

### CODE REVIEW

**Comment by anthony-malkoun** [comment]
@christopher-hopper what was the reasoning behind this going in `tide_site`? It feels more a `tide_core` thing to me.

**Comment by christopher-hopper** [comment]
> @christopher-hopper what was the reasoning behind this going in `tide_site`? It feels more a `tide_core` thing to me.

@anthony-malkoun 
I looked for where I found Tide specific Menu generation happening and located it there. It was said that "Add menu" needed to be hidden because Tide generated its own Menu for each site. 

I saw the service 'tide_site.menu_autocreate' and thought to keep these functions together. If Tide isn't making its own menus, I figured "Add menu" didn't need to be controlled.

**Comment by anthony-malkoun** [comment]
> 



> > @christopher-hopper what was the reasoning behind this going in `tide_site`? It feels more a `tide_core` thing to me.
> 
> @anthony-malkoun I looked for where I found Tide specific Menu generation happening and located it there. It was said that "Add menu" needed to be hidden because Tide generated its own Menu for each site.
> 
> I saw the service 'tide_site.menu_autocreate' and thought to keep these functions together. If Tide isn't making its own menus, I figured "Add menu" didn't need to be controlled.

Huh, TIL. Good sleuthing @christopher-hopper

**Review by anthony-malkoun — APPROVED** [review]
Cheers for cleaning up some stuff too... more reason to get our linting/CS updated.

File: tide_site.info.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ dependencies:
   - dpc-sdp:tide_media
   - entity_browser:entity_browser
   - entity_browser:entity_browser_entity_form
+  - create_menus_permission:create_menus_permission
```
Comment (christopher-hopper) [inline tide_site.info.yml:20]: @anthony-malkoun should we make this "Create menus permission" a required dependency?

Alternative would be to make optional, allowing sites to use more fine-grained alternatives or nothing at all. You could show something in the Drupal Status Report as a recommendation if the module was not installed. 

Making it a required dependency for now. If an issue does arise within the user-base, we could make it optional.

File: src/Commands/TideSiteCommands.php [hunk 2]
```diff
@@ -32,7 +35,7 @@ public function siteEnvDomainUpdate() {
     try {
       $environment = getenv('LAGOON_GIT_BRANCH');
```
Comment (anthony-malkoun) [inline src/Commands/TideSiteCommands.php:36]: Unrelated, but this is the wrong check. Should be on `LAGOON_ENVIRONMENT_TYPE`.

File: tide_site.info.yml (hunk 1 above)
Comment (anthony-malkoun) [reply tide_site.info.yml:20]: Required dependency is fine here I think.

**Review by anthony-malkoun — APPROVED** [review]
👍

**Pull request merged**